lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -281,7 +291,9 @@ newStakePoolLayer nl db@DBLayer {..} worker = do
         case r of
             Left e@(PastHorizon{}) ->
                 throwE (ErrListPoolsPastHorizonException e)
-            Right r' -> pure r'
+            Right r' -> pure  
+                $ ApiListStakePools r' $ Just $ ApiT
+                $ Iso8601Time $ posixSecondsToUTCTime lastGC
```

Comment:
Given that this is always `Just`, it seems that we might run the possibility of returning the start of the UNIX epoch (`1970-01-01T00:00:00Z`) in the case where there hasn't yet been a garbage collection, unless I'm mistaken?

As mentioned above, would recommend changing the type of `readLastMetadataGC` to return `Maybe POSIXTime`. This maps more cleanly onto the API representation IMO.

Reply:
I don't think it does. The frontend should never really see this value, except when there is a race, in which case this value is still harmless. Whenever you query the stake pools, there should have been a GC before. The frontend isn't supposed to know or react to the case where a GC has not happened yet.

Reply:
Maybe is a very inexpressive type. So if we really have a problem with the null-value here, then we need to go for a proper sum type. But I don't think there's value in it. My opinion about this isn't too strong though.

Reply:
Well, we already use `Maybe` in several places in the wallet API to reflect the lack of a value.

In JSON, we often represent `Nothing` as a _missing field_, with a suitable comment in the API documentation.

For example, consider the optional `retirement` field of `ApiStakePool`:

```yaml
ApiStakePool: &ApiStakePool   
  type: object      
  required:         
    - id  
    ...
  properties:       
    id: *stakePoolId
    ...
    retirement: *stakePoolRetirement  

x-stakePoolRetirement: &stakePoolRetirement
  <<: *epochInfo
  description: |
      The epoch in which a stake pool retires.

      May be omitted if the wallet hasn't yet found a retirement certificate
      for this stake pool.
```

For stake pools that don't have a retirement certificate, this field is omitted, and the serialization is unambiguous.

Of course we _could_ use a dedicated sum-type here, but I think `Maybe` is also fine, assuming we add an appropriate comment to the API docs.

Reply:
There are four cases we might want to encode as Nothing:

1. there is no concept of delisted pools (jormungandr)
2. the current fetch-metadata setting is set to `none` (currently, it will return the last GC time regardless)
3. the GC thread has never run yet (this shouldn't really be possible to observe)
4. the GC thread has just been restarted (daedalus already needs to consider a short grace period, before querying the pools again after GC restart)

That's quite ambiguous, but tbh, deadalus doesn't really need any of this information. It already has all the info.

Reply:
I guess if we do this, we also want to expose the GC status as its own endpoint, so daedalus can poll the status until the thread has restarted.